elasticdl/python/common/k8s_job_monitor.py

```diff
@@ -0,0 +1,199 @@
+import time
+import traceback
+
+from kubernetes import client, config
+
+from elasticdl.python.common.constants import PodStatus
+from elasticdl.python.common.k8s_client import Client
+from elasticdl.python.common.log_utils import default_logger as logger
+
+MAX_READ_POD_RETRIES = 6
+
+
+def print_tail_log(log, tail_num):
+    if log is not None:
+        log_lines = log.split("\n")
+        print("\n".join(log_lines[-1 * tail_num:]))
+
+
+class PodMonitor:
+    def __init__(self, namespace, pod_name, use_kube_config=True):
+        """
+        k8s Pod Monitor. During data transformation, we may launch a pod
+        on k8s cluster to perform data analysis and need to monitor the
+        pod execution status.
+
+        Args:
+            namespace: The name of the Kubernetes namespace where the pod
+                will be created.
+            pod_name: Pod name, should be unique in the namespace.
+            use_kube_config: If true, load the cluster config from
+                ~/.kube/config. Otherwise, if it's in a process running in
+                a K8S environment, it loads the incluster config.
+        """
+        self.namespace = namespace
+        self.pod_name = pod_name
+        self.core_api = client.CoreV1Api()
+
+        try:
+            if use_kube_config:
```

Comment:
Can we reuse existing code?

Reply:
I move the `get_pod` and `get_pod_log` to `k8s_client`. Then we don't need those codes.